### PR TITLE
Improve testing infrastructure and enable analyze mode in production builds

### PR DESCRIPTION
**Pull request by morsdyce**
1. Enable analyze mode in production builds so it's possible to see only assets which reach production
2. Add default configuration on where to find tests.
3. Add support for testing components with scss imports
4. Add enzyme snapshot serializer for better enzyme snapshots
5. Add automatic mocking for react-intl
6. Upgrade jest to latest version so we can use all the shiny new features
7. Remove some of the old sample code from template as it is no longer relevant in this location and interrupts testing react-scripts

### CODE REVIEW

File: packages/react-scripts/config/webpack.config.prod.js [hunk 1]
```diff
@@ -36,6 +36,9 @@ const publicUrl = publicPath.slice(0, -1);
 // Get environment variables to inject into our app.
 const env = getClientEnvironment(publicUrl);
 
+const BundleAnalyzerPlugin = require('webpack-bundle-analyzer').BundleAnalyzerPlugin;
+const isAnalyzeMode = process && process.env && process.env.ANALYZE;
```
Comment (adamkleingit) [inline packages/react-scripts/config/webpack.config.prod.js]: Maybe call it WEBPACK_ANALYZE, or something more specific?

File: packages/react-scripts/config/jest/react-intlMock.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+'use strict';
+
+const React = require('react');
+const intlShape = require('react-intl').intlShape;
+
+const addLocaleData = () => {};
+const FormattedMessage = () => React.createElement('div');
```
Comment (adamkleingit) [inline packages/react-scripts/config/jest/react-intlMock.js:7]: Does this mean that if we change the key or params the test will still pass if we do matchSnapshot?

File: packages/react-scripts/config/paths.js [hunk 3]
```diff
@@ -53,7 +53,6 @@ module.exports = {
   appIndexJs: resolveApp('src/index.js'),
   appPackageJson: resolveApp('package.json'),
   appSrc: resolveApp('src'),
-  testSrc: resolveApp('tests'),
```
Comment (adamkleingit) [inline packages/react-scripts/config/paths.js:56]: Why remove it?

File: packages/react-scripts/config/jest/react-intlMock.js (hunk 2 above)
Comment (morsdyce) [reply packages/react-scripts/config/jest/react-intlMock.js:7]: No, in the snapshots you still see the component name and props.
This is how it looks in the snapshots:

```jsx
<FormattedMessage
        id="common.alerts"
      />
```

File: packages/react-scripts/config/paths.js (hunk 3 above)
Comment (morsdyce) [reply packages/react-scripts/config/paths.js:56]: It's not used anywhere anymore, will be easier to get updates from upstream if we don't leave things that aren't needed anymore.

**Pull request merged**